utils.py

```diff
@@ -176,52 +175,89 @@ def combine_vcf(combined_vcf, vcfs, duplicate_handling_mode = COMBINE_KEEP_ALL_D
-        pysam.tabix_index(vcf, force=True, preset='vcf')
-        return gz_vcf
-    elif mode == 2:
-        suffix_index = vcf.rfind('.vcf')
-        sorted_vcf = vcf[:suffix_index] + ".sorted" + vcf[suffix_index:]
-        gz_vcf = "{}.gz".format(sorted_vcf)
-
-        sort_command = [SORT_VCF, vcf]
-        logger.info('sorting {}'.format(vcf))
-        if (not overwrite) and os.path.isfile(sorted_vcf):
-            raise ValueError("{} exists".format(sorted_vcf))
-        with open(sorted_vcf, "w") as sorted_out:
-            run_shell_command(sort_command, cmd_stdout=sorted_out, cmd_stderr=sys.stderr)
-        logger.info('compressing {}'.format(sorted_vcf))
-        if (not overwrite) and os.path.isfile(gz_vcf):
-            raise ValueError("{} exists".format(gz_vcf))
-        with open(gz_vcf, "w") as out:
-            run_shell_command([BGZIP, "--force", "--stdout", sorted_vcf], cmd_stdout=out, cmd_stderr=sys.stderr)
-        index_vcf_gz(gz_vcf)
-        return gz_vcf
-    else:
-        raise ValueError
+    gz_vcf = "{}.gz".format(vcf)
+    sorted_vcf = "{}.sorted".format(vcf)
+
+    sort_command = [SORT_VCF, vcf]
+    with open(sorted_vcf, "w") as sorted_out:
+        run_shell_command(sort_command, cmd_stdout=sorted_out, cmd_stderr=sys.stderr)
+    os.rename(sorted_vcf, vcf)
+    pysam.tabix_index(vcf, force=True, preset='vcf')
+    return gz_vcf
+
+
+def write_vcf(lines, vcf):
+    """Create a file from the provided list"""
+
+    with open(vcf, "w") as vcf_handle:
+        vcf_handle.write('\n'.join(lines))
+    return vcf
+
+
+def get_equivalent_variant(variant, vcf):
+    """Return the variant in a vcf closest to a variant"""
+
+    equivalent_variant = None
+
+    with versatile_open(vcf, "r") as vcf_handle:
+        min_dist = 100
+
+        for line in vcf_handle.readlines():
+            line_split = str(line).strip().split()
+
+            if line_split[0][0] == "#":
+                continue
+
+            if line_split[0] == variant[0]:
+                dist = abs(int(line_split[1])-int(variant[1]))
+
+                if dist < min_dist:
+                    min_dist = dist
+                    equivalent_variant = line_split[:]
+
+    return equivalent_variant
+
+
+def make_clean_vcf(vcf, path=None):
+    """Make a clean vcf retaining essential fields"""
+
+    vcf_path = os.path.split(vcf)
+    vcf_base = os.path.splitext(vcf_path[1])
+
+    if not path:
+        path = vcf_path
+
+    clean_vcf = os.path.join(path, vcf_base[0] + ".clean" + vcf_base[1])
+
+    clean_vcf_handle = open(clean_vcf, "w")    
+
+    with versatile_open(vcf, "r") as vcf_handle:
+        for line in vcf_handle.readlines():
+            line_strip = line.strip()
+            line_split = line_strip.split()
+
+            if line_strip[0] == "#":
+                clean_vcf_handle.write(line_strip+'\n')
+                continue
+            else:
+                GT = "0/1" if line_split[-1] == "./." else line_split[-1]
```

Comment:
em, are we sure to convert all `./.` to `0/1`?

Reply:
If the user doesn't specify GT, shouldn't we should default to something? Do you think there's a better option?